.github/workflows/windows.yml

```diff
@@ -49,20 +49,20 @@ jobs:
 
       - name: pip-packages
         run: |
-          pip install numpy cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool hypothesis "scipy-openblas32<=0.3.23.293.2"
+          pip install numpy cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich_click click doit pydevtool hypothesis "scipy-openblas32<=0.3.23.293.2"
 
       - name: Build
         run: |
           python dev.py build --with-scipy-openblas
 
       - name: Test
         run: |
-          python dev.py test -j2
+          python dev.py test -j2 -- --durations=0 --durations-min=0.25 --fail-slow=1.0
```

Comment:
FWIW, I think the original experiment of adding the one CI job with `pytest-fail-slow` below, with a 5-second global threshold to catch truly egregious single test cases, was a good thing.

That said, I'm less keen on adding another CI job and a second timing threshold to be aware of. I think this is going to be subjective of course, but it seems to me that we'd just be discouraging one form of total test time creep, but there are many others like overparametrization that would add up as well.

At some point, the code reviewer will have to look the tests over on a case by case basis to see what makes sense and what doesn't. I think the 5 second cutoff for "you should take a look at that" make sense, but an additional 1 second cutoff is starting to make the automatic catching a bit complex/picky for my taste. It may be preferable to have a single 1.1 second test vs. a parametrized test with 37 0.8 second cases of course, and I don't know if I'd want the CI weighing in on that this way.

That said, I don't care that much. In general, I'd probably opt for just putting the exception markers on anything that gets flagged in `spatial` rather than hiding the tests in the `full`/slow mode, which I often don't use when iterating locally.

Reply:
The total time of the `spatial` tests on the list is only 10s, so that seems OK to me to make an exception and give them 5s.